### ISSUE TITLE
Analytics Implementation for [Profile 2.0]

### ISSUE THREAD

**Issue by mshea0606** [department-of-veterans-affairs/va.gov-team#9488]
# Google Analytics Implementation or QA Request

## What this form is for

<details>
<summary>More Information</summary>
<div>
    <br>
<p>Use this template to request Google Tag Manager and Google Analytics implementation or QA for your product. Please fill out all brackets.</p>
    
> _Please see the following information for implementation timeline:_
> New Teams: Implementation will usually be completed in 4 sprints:
> 
> - Sprint 1: Discovery/Orientation
> - Sprint 2: FE Implementation
> - Sprint 3: Tagging Implementation
> - Sprint 4: QA
> 
> All other teams: Implementation will usually be completed in 2 sprints
> 
> - Sprint 1: Discovery and FE Implementation
> - Sprint 2: Tagging Implementation and QA

For more information, please check out the [Google Tag Manager resources](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/analytics/google-analytics/readme.md).

An example of a completed request template can be found [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7199). A fully completed issue can help speed up the implementation process.
</div>
</details>

## Description
- This is an Analytics/GA request on behalf of the Authenticated Experience team for Profile 2.0.
[Profile 2.0 Product Outline](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/identity-personalization/profile/Combine%20Profile%20and%20Account)

- Who is this request for?
Matt Shea and Samara Strauss

- What team/project is this work for?

Team: Authenticated Experience Team 
Project: Profile 2.0 
The goals of this project are to:
1) Combine profile and account pages to reduce user confusion and improve way finding
2) Enabled scalability of profile for additional features
3) Increase personalization
4) Improve design
5) Decrease time/effort to find and complete tasks

- Provide the Domain where Google Analytics needs to be implemented
https://staging.va.gov/profile  (No PII info in URL/title tag)

- Please provide any upcoming deadlines or important milestones
1) Build should be complete by the end of AE-Sprint 23 (Mid June),
2) Estimated launch will be in July

- Typical Website Interactions to Track
   - *Conversions - Understanding the User Flow:* _Provide URLs that are part of the application from introduction page to successful form completion; conditional pages that are visible depending on an applicant's answer should be marker accordingly._
- https://staging.va.gov/profile 
    
- *Other User Interactions*: _Provide details on other interactions and metrics, like click to help text links, that are helpful in understanding user behavior._
- Not sure how to answer this

Sample Users when authentication is required _**do not put staging credentials in your `va.gov-team` ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository**_

- Test Users to replicate the event _**do not put staging credentials in your `va.gov-team` ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository**_
- [Profile 2.0 Staging Users - Test cases](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/Combine%20Profile%20and%20Account/QA/Staging%20Users%20-%20Test%20Cases.md)

- Additional Comments
To view Profile 2.0:
Sign in to VA.gov (All staging users will see new the new profile)
Go to the user profile

What we want to track for this feature:
[Analytics needs for Profile 2.0](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/Combine%20Profile%20and%20Account/analytics/Analytics%20needs%20for%20profile%202.0.md)

## Acceptance Criteria
- [ ] QA has been completed
- [ ] VFS team completes brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Definition of Done
- [ ] All appropriate issue tagging is completed
- [ ] All AC completed

**Comment by jonwehausen** [comment]
### Notes from Todays Review
1) Mini audit all existing tracking
  > Confirming where any dataLayer cleanup needs to happen - determine where existing updates need to happen and where new events need to be put in (i.e connected apps)
  > Confirming where any syntax tagging cleanup  
[Audit trail here](https://docs.google.com/spreadsheets/d/1jxdvV2v3V-E_9431ZBjwJVgSGhX8ykPtZCOWKPPQSdA/edit#gid=0)
2) Break out Contact Information dashboard into separate issue
3) Brian to test out authentication events when test users are available

**Comment by bsmartin-ep** [comment]
@jonwehausen / @mshea0606  -

I believe we'll need the following dataLayer and GTM changes.  Let me know if you want to have a quick meeting to discuss this.

## dataLayer

### 1. "How many users have apps connected"

Since this is new, we'll need to get a boolean pushed to the dataLayer (and later connected to a new Custom Dimension), both when they **disconnect an app** and when **they login (or the profile loads)**.  @mshea0606 - do we know their Connected Apps status when they login or not until they access the profile?

#### Login (or Profile load);
```
<script>
dataLayer = [{
  'user-has-connected-apps': true 
}];
</script>
```
#### App Disconnect

```
{
  'event': 'profile-navigation',
  'profile-action': 'disconnect-button',
  'profile-section': 'connected-accounts',
  'user-has-connected-apps': true|false,
}
```

### 2. Existing _account-_ events and dataLayer variables should be _profile-_

Events like:

```
{
  'event': 'account-navigation',
  'account-action': 'view-link',
  'account-section': 'vets-faqs',
}
```
...need to be consolidated into _profile-_ like...

```
{
  'event': 'profile-navigation',
  'profile-action': 'view-link',
  'profile-section': 'vets-faqs',
}
```
We'll update our GTM lookup tables and GA dashboards to use these new values.

@mshea0606 -  Do you know if this event is in use?

```
'event': 'account-terms-transaction'
```

## Google Tag Manager

### 3. Update _Profile Form Submit_ tag to use lookup table.  

Right now we are getting the actual event in our action like...

```Profile - Form Submit - home-address```

...when this should go through the lookup table like other Profile tags...

```Profile - Form Submit - Home Address```

This will be a GTM update and dashboard update done by Insights and Analytics.

---
Anything to add @jonwehausen ?

**Comment by Samara-Strauss** [comment]
Just confirming next steps:

- FE needs to make the changes you outlined above.
- We'll have you guys take a look at that stuff when it's done to make sure it looks good.

Yes?

**Comment by mshea0606** [comment]
Recap from meeting:
- A user's connected apps status is loaded when a user clicks the "Connected Apps" section of the Profile
- Samara to finalize OKR/KPI's
- Once OKR/KPI's are finalized, regroup with Analytics team to determine if we will do GA or Domo

**Comment by mshea0606** [comment]
@erikphansen @sanlouise  Regarding the comment above...Do you know if this event is in use?

'event': 'account-terms-transaction' @bsmartin-ep @jonwehausen

**Comment by bsmartin-ep** [comment]
Hi @Samara-Strauss / @mshea0606 -

Just checking in on the status of this item:

- A user's connected apps status is loaded when a user clicks the "Connected Apps" section of the Profile

I just checked on User 1 and am not yet seeing this in the dataLater:

```
{
  'user-has-connected-apps': true|false
}
```

Let us know when you are ready for us to re-test.

Thanks!

**Comment by Samara-Strauss** [comment]
I'm so sorry — I didn't realize this never got added back in. I've asked in the connected apps channel in Slack and will keep you posted.

**Comment by Samara-Strauss** [comment]
Is this done? I thought the connected apps set up still needed to happen.

Also, are the new dashboards ready? If so, we'd love to walk through where to find them with your team.

**Comment by Samara-Strauss** [comment]
Got it. Thanks!

**Comment by bsmartin-ep** [comment]
@jonwehausen -

I'm seeing the DL stuff for connected apps:

![image](https://user-images.githubusercontent.com/62304138/88550284-63fed580-cfef-11ea-813e-52b087b4cc6a.png)

One question - this is just a data layer push (no event or pageview attached to it), so if I add it to our _Base Configuration_ GA settings var as a CD it won't be actually sent into GA until the next event/Pageview.

Should we ask them to generate an event around this?

**Comment by bsmartin-ep** [comment]
@jonwehausen -

## Connected Apps QA

Description|Data Layer|Tag
--|--|--
Connected Apps Started|![image](https://user-images.githubusercontent.com/62304138/88867414-69207800-d1db-11ea-8181-bb45bb9e89e5.png)|![image](https://user-images.githubusercontent.com/62304138/88867447-7fc6cf00-d1db-11ea-9b9e-03a5e6b4d95c.png)
Connected Apps Retrieved|![image](https://user-images.githubusercontent.com/62304138/88867468-8a816400-d1db-11ea-93d4-a952d9952007.png)|![image](https://user-images.githubusercontent.com/62304138/88867488-99681680-d1db-11ea-96e3-d1fde8bc18a5.png)
Connected Apps Failed<br>(simulated by blocking URL in devtools)|![image](https://user-images.githubusercontent.com/62304138/88867673-290dc500-d1dc-11ea-937f-8cfc3cee2069.png)|![image](https://user-images.githubusercontent.com/62304138/88867688-35921d80-d1dc-11ea-888b-35a3f439ad56.png)

I think this is everything, Jon.  Want Ben to re-test everything?

**Comment by bsmartin-ep** [comment]
FYI @Samara-Strauss / @mshea0606 -

We published our GTM workspace **\*to staging\*** so Connected Apps events _etc._ should be appearing in non-prod GA:

![image.png](https://images.zenhubusercontent.com/5e73c8e4b62b077da52f1c14/d017a368-fee0-400e-9634-35868d831939)

https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w178298118p187683383/explorer-segmentExplorer.segmentId=analytics.eventLabel&explorer-table.plotKeys=%5B%5D&explorer-table.filter=profile-&explorer-table.rowStart=0&explorer-table.rowCount=25/

**Comment by bsmartin-ep** [comment]
@markgreenburg / @sanlouise -

As promised, here is a dataLayer spec for Connected Apps disconnect:

```
{
  'event': 'profile-disconnect-connected-app-started'
}

{
  'event': 'profile-disconnect-connected-app-failed',
  'error-key': '00_error message'
}

{
  'event': 'profile-disconnect-connected-app-successful',
  'user-has-connected-apps': true|false
}
```
### Would you be able to do us a HUGE favor while making this change?

We would like the Get Connected Apps / Direct Deposit events to be changed to match our normal past-tense taxonomy.

So:
`profile-get-connected-apps-failure` would need to become `profile-get-connected-apps-failed`
`profile-get-direct-deposit-failure` would need to become `profile-get-direct-deposit-failed`

Let us know if this is possible and we can update our Dashboards etc to use the new event.

Thanks!

cc @jonwehausen

**Comment by sanlouise** [comment]
@bsmartin-ep Unless @markgreenburg sees an issue with the updates, on my end this would be absolutely no problem! I created [this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/12399) and will ping you Brian once it's merged.

**Comment by bmcgrady-ep** [comment]
@mshea0606 / @jonwehausen / @Samara-Strauss 

It looks like we will need the following dataLayer and GTM changes. Currently, no events are firing for Edit Account clicks and Add Account clicks in Profile 2.0 Direct Deposit.

## dataLayer
### 1.  In Direct Deposit - Clicks Add Account
![image](https://user-images.githubusercontent.com/66263904/90542595-3530d680-e139-11ea-94b0-1337c73c8181.png)
```
{
      'event': 'profile-navigation',
      'profile-action': 'add-link',
      'profile-section': 'direct-deposit-information'
}
```

### 2. In Direct Deposit - Clicks Edit Account
![image](https://user-images.githubusercontent.com/66263904/90542695-5db8d080-e139-11ea-8bbe-4aa673c6ba06.png)
```
{
      'event': 'profile-navigation',
      'profile-action': 'edit-link',
      'profile-section': 'direct-deposit-information'
}

```

**Comment by Samara-Strauss** [comment]
Also, adding this info which I shared with Ben in Slack and don't want to lose:

- Right now in Profile 1.0, there are three edit links for direct deposit, but they all open up the same form.
- If the person is eligible for direct deposit but has not added any direct deposit information, a link shows under each of those headings: "Please add a bank name," "Please add an account number," "Please add an account type"
- You can see on our GA dashboard that we track all three "Edit" and all three "Add" links. **We do not need to track these going forward because, as Ben showed above, there is only one Edit and one Add link in the new design.

**Comment by Samara-Strauss** [comment]
@bsmartin-ep @bmcgrady-ep — @sanlouise made the data layer updates request above, so you should be good to go to move this ticket forward.

**Comment by Samara-Strauss** [comment]
@bsmartin-ep @bmcgrady-ep direct deposit is working again in Profile 2.0, so you can review Sandra's updates mentioned above.

**Comment by bmcgrady-ep** [comment]
### DataLayer Changes QA
Form Step / Interaction | GTM Screenshot | Test Status
-- | -- | --
Clicks Add Account in Direct Deposit | ![image](https://user-images.githubusercontent.com/66263904/92026188-7f0ee480-ed15-11ea-8245-a1b1d9de548f.png) | ✔️ **PASS**
Clicks Edit Account in Direct Deposit | ![image](https://user-images.githubusercontent.com/66263904/92026589-1a07be80-ed16-11ea-9f85-349efc7b60ae.png) | ✔️ **PASS**

**Comment by Samara-Strauss** [comment]
Great! We can close this out after our handoff meeting later today if everything looks good. Thank you so much for all your hard work!

**Comment by bmcgrady-ep** [comment]
@Samara-Strauss / @mshea0606 

Thank you for taking the time to meet with us today. Here are the links to the different dashboards we discussed today.
- [Connected Apps Dashboard](https://analytics.google.com/analytics/web/?authuser=0#/dashboard/LtprxgtsTZu4Mxt-16iKkA/a50123418w178298118p176884211/) [Staging]
- [Address Validation Dashboard](https://analytics.google.com/analytics/web/#/dashboard/pq_-PrkvQleUdCBbV7eq7Q/a50123418w177519031p176188361/) [Production]
- [Direct Deposit Dashboard](https://analytics.google.com/analytics/web/?authuser=0#/dashboard/naG_-UneTxy50WvvIH0GGQ/a50123418w177519031p176188361/) [Production]
- [Direct Deposit Dashboard](https://analytics.google.com/analytics/web/#/dashboard/77_LQy01QnSsGhdEkf3Qlw/a50123418w178298118p176884211/) [Staging]
- [Account Security Dashboard](https://analytics.google.com/analytics/web/#/dashboard/fybwEuuTRCmaQjsUVDBGJA/a50123418w178298118p176884211/) [Staging]
- [Account Security Dashboard](https://analytics.google.com/analytics/web/?authuser=0#/dashboard/AuydfL6WR7WS_nXqNhC5GA/a50123418w177519031p176188361/) [Production]
- [Contact Information Dashboard](https://analytics.google.com/analytics/web/?authuser=0#/dashboard/3Z1vT6SlQqevdav1s72QwA/a50123418w177519031p176188361/) [Production]

As we discussed, the dashboards that are in both staging and production only have data in staging, but will start to have data in production when Profile 2.0 is live. 

Let me know if you have any questions!

**Comment by Samara-Strauss** [comment]
Great! Thank you so much for all of these.

**Comment by Samara-Strauss** [comment]
Also, I think we can close out this ticket *unless* you want to wait for the Domo work to be done as part of it.

**Comment by Samara-Strauss** [comment]
Actually, one question, @bmcgrady-ep — what is the difference between the account settings dashboard in staging and the one in production? Trying to figure out which one I should bookmark/monitor after we go live.

**Comment by bmcgrady-ep** [comment]
@Samara-Strauss there isn't any difference between the account security dashboards. The dashboards are built for the Profile 2.0 account security section, so currently data is only in the staging dashboard.

I think we can close out this ticket whenever you would like to. We can track the DOMO KPI dashboard in this [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/12621).

**Comment by Samara-Strauss** [comment]
Great! Thanks for clarifying.

Since we're tracking the Domo work elsewhere, we can call this #done!
